### ISSUE TITLE
a number of push tests (4?) started to fail at 20210720 on windows with some recent annex changes

### ISSUE THREAD

**Issue by yarikoptic** [datalad/datalad#5811]
```shell
$> git grep datalad.core.distributed.tests.test_push.test_gh1426 | grep release | grep windows                            
...
cron-20210719/build-windows.yaml-321-69466103-failed/test-datalad (release)/21_Run datalad tests.txt:2021-07-19T04:06:18.7313658Z datalad.core.distributed.tests.test_push.test_gh1426 ... ok
cron-20210720/build-windows.yaml-322-69466103-failed/3_test-datalad (release).txt:2021-07-20T04:07:27.9583842Z datalad.core.distributed.tests.test_push.test_gh1426 ... ERROR
...
```
see e.g. https://github.com/datalad/git-annex/runs/3130519727?check_suite_focus=true

so might be some regression or change in behavior of annex (ping @joeyh)

edit, forgot to paste a full list:
```
(git)smaug:…lds/2021/07[master]cron-20210722/build-windows.yaml-324-69466103-failed
$> git grep 'ERROR:'                                                                                                      
1_test-datalad (master).txt:2021-07-22T04:51:00.2969878Z ERROR: File C:\Users\runneradmin\AppData\Local\Temp\datalad_temp_l3q9itaa\providers\exists.cfg already exists, choose another name
1_test-datalad (master).txt:2021-07-22T04:51:00.3083408Z ERROR: Provided regular expression doesn't match original url.  Please re-enter
1_test-datalad (master).txt:2021-07-22T06:21:28.4388312Z ERROR: datalad.core.distributed.tests.test_push.test_push_git_annex_branch_many_paths_same_data
1_test-datalad (master).txt:2021-07-22T06:21:28.4422379Z ERROR: datalad.core.distributed.tests.test_push.test_push_matching
1_test-datalad (master).txt:2021-07-22T06:21:28.4451783Z ERROR: datalad.core.distributed.tests.test_push.test_gh1426
1_test-datalad (master).txt:2021-07-22T06:21:28.4482878Z ERROR: datalad.core.distributed.tests.test_push.test_gh1763
...
```

**Comment by joeyh** [comment]
What makes you think this has anything to do with git-annex?

-- 
see shy jo

**Comment by yarikoptic** [comment]
they started to fail on the day 20210720 whenever they passed a day before while there were no changes to datalad itself (it was a released version and we didn't release since `Jun 27`. Having said that some other dependencies could have changed, didn't check, but they are less likely to introduce such a chance

**Comment by adswa** [comment]
I fail to install the most recent version of git-annex on the Windows box... Can you remind me how to do that? (sorry :( I think I ask this every 4 months :( ) The most recent I got so far is 8.20210429, but I bet I used the wrong installer methods. Although I also couldn't find 8.20210720 in https://datasets.datalad.org/datalad/packages/windows/?

**Comment by adswa** [comment]
okay, I am able to reproduce the failures with this WTF:
<details>

```
C:\Users\datalad\repos\datalad>datalad wtf
# WTF
## configuration <SENSITIVE, report disabled by configuration>
## credentials
  - keyring:
    - active_backends:
      - Windows WinVaultKeyring
      - Windows RegistryKeyring
      - EncryptedKeyring with not defined v.1.0 at C:\Users\datalad\AppData\Local\Python Keyring\wincrypto_pass.cfg
      - PlaintextKeyring with no encyption v.1.0 at C:\Users\datalad\AppData\Local\Python Keyring\keyring_pass.cfg
    - config_file: C:\Users\datalad\.config\python_keyring\keyringrc.cfg
    - data_root: C:\Users\datalad\AppData\Local\Python Keyring
## datalad
  - full_version: 0.14.4.dev507-g66fd
  - version: 0.14.4.dev507
## dataset
  - branches:
    - addurls-key-adjusted@bef9b64
    - appveyor@a3e17b9
    - audit-get-annexed-files@c886ac1
    - bf-2575@15dc9f9
    - bf-4359@9104ffb
    - bf-4822@40469d2
    - bf-5211@00bba27
    - bf-winhome@1bdb0c2
    - bf-winhome2@f8c5a90
    - bf-winhomebackup@7c89355
    - maint@83463ee
    - maintbackup@a92ec4c
    - master@66fd8be
    - opt-annexrepo@af81741
    - tmp-gitmoduleswin@3f57540
    - tst-adjustedbranch@97831a5
    - tst-lesstestrepos@3cd62d7
    - tst-pathlengths@a7adf3f
    - tst-testrepos-appveyor@a37d111
    - win-addurltests@38a9b51
    - win-addurltestsbackup@568616e
    - win-addurltestsbackup2@304692c
    - win-gitannex@9bfbe7f
    - win-quoting@085bad2
    - win-replace@3a49df7
    - win-replacebackup@4a09de1
  - id: None
  - metadata: <SENSITIVE, report disabled by configuration>
  - path: C:\Users\datalad\repos\datalad
  - repo: GitRepo
## dependencies
  - annexremote: 1.5.0
  - appdirs: 1.4.4
  - boto: 2.49.0
  - cmd:7z: 19.00
  - cmd:annex: 8.20210715-g3b5a3e168
  - cmd:bundled-git: UNKNOWN
  - cmd:git: 2.32.0.windows.2
  - cmd:system-git: 2.32.0.windows.2
  - cmd:system-ssh: for
  - humanize: 3.2.0
  - iso8601: 0.1.14
  - keyring: 23.0.0
  - keyrings.alt: 4.0.2
  - msgpack: 1.0.2
  - requests: 2.25.1
  - wrapt: 1.12.1
## environment
  - PATH: C:\Program Files\Docker\Docker\Resources\bin;C:\WINDOWS\system32;C:\WINDOWS;C:\WINDOWS\System32\Wbem;C:\WINDOWS\System32\WindowsPowerShell\v1.0\;C:\WINDOWS\System32\OpenSSH\;C:\ProgramData\chocolatey\bin;C:\Program Files (x86)\Gpg4win\..\GnuPG\bin;C:\Program Files\Git\cmd;C:\Users\datalad\AppData\Local\Programs\Python\Python39\Scripts\;C:\Users\datalad\AppData\Local\Programs\Python\Python39\;C:\Users\datalad\AppData\Local\Microsoft\WindowsApps;C:\Program Files (x86)\GnuWin32\bin;C:\Program Files\7-Zip;
## extensions
  - container:
    - description: Containerized environments
    - entrypoints:
      - datalad_container.containers_add.ContainersAdd:
        - class: ContainersAdd
        - load_error: None
        - module: datalad_container.containers_add
        - names:
          - containers-add
          - containers_add
      - datalad_container.containers_list.ContainersList:
        - class: ContainersList
        - load_error: None
        - module: datalad_container.containers_list
        - names:
          - containers-list
          - containers_list
      - datalad_container.containers_remove.ContainersRemove:
        - class: ContainersRemove
        - load_error: None
        - module: datalad_container.containers_remove
        - names:
          - containers-remove
          - containers_remove
      - datalad_container.containers_run.ContainersRun:
        - class: ContainersRun
        - load_error: None
        - module: datalad_container.containers_run
        - names:
          - containers-run
          - containers_run
    - load_error: None
    - module: datalad_container
    - version: 1.1.2
## git-annex
  - build flags:
    - Assistant
    - Webapp
    - Pairing
    - TorrentParser
    - MagicMime
    - Feeds
    - Testsuite
    - S3
    - WebDAV
  - dependency versions:
    - aws-0.22
    - bloomfilter-2.0.1.0
    - cryptonite-0.26
    - DAV-1.3.4
    - feed-1.3.0.1
    - ghc-8.8.4
    - http-client-0.6.4.1
    - persistent-sqlite-2.10.6.2
    - torrent-10000.1.1
    - uuid-1.3.13
    - yesod-1.6.1.0
  - key/value backends:
    - SHA256E
    - SHA256
    - SHA512E
    - SHA512
    - SHA224E
    - SHA224
    - SHA384E
    - SHA384
    - SHA3_256E
    - SHA3_256
    - SHA3_512E
    - SHA3_512
    - SHA3_224E
    - SHA3_224
    - SHA3_384E
    - SHA3_384
    - SKEIN256E
    - SKEIN256
    - SKEIN512E
    - SKEIN512
    - BLAKE2B256E
    - BLAKE2B256
    - BLAKE2B512E
    - BLAKE2B512
    - BLAKE2B160E
    - BLAKE2B160
    - BLAKE2B224E
    - BLAKE2B224
    - BLAKE2B384E
    - BLAKE2B384
    - BLAKE2BP512E
    - BLAKE2BP512
    - BLAKE2S256E
    - BLAKE2S256
    - BLAKE2S160E
    - BLAKE2S160
    - BLAKE2S224E
    - BLAKE2S224
    - BLAKE2SP256E
    - BLAKE2SP256
    - BLAKE2SP224E
    - BLAKE2SP224
    - SHA1E
    - SHA1
    - MD5E
    - MD5
    - WORM
    - URL
    - X*
  - operating system: mingw32 x86_64
  - remote types:
    - git
    - gcrypt
    - p2p
    - S3
    - bup
    - directory
    - rsync
    - web
    - bittorrent
    - webdav
    - adb
    - tahoe
    - glacier
    - ddar
    - git-lfs
    - httpalso
    - borg
    - hook
    - external
  - supported repository versions:
    - 8
  - upgrade supported from repository versions:
    - 2
    - 3
    - 4
    - 5
    - 6
    - 7
  - version: 8.20210715-g3b5a3e168
## location
  - path: C:\Users\datalad\repos\datalad
  - type: dataset
## metadata_extractors
  - annex (datalad 0.14.4):
    - distribution: datalad 0.14.4
    - load_error: None
    - module: datalad.metadata.extractors.annex
    - version: None
  - audio (datalad 0.14.4):
    - distribution: datalad 0.14.4
    - load_error:
    - module: datalad.metadata.extractors.audio
  - datacite (datalad 0.14.4):
    - distribution: datalad 0.14.4
    - load_error: None
    - module: datalad.metadata.extractors.datacite
    - version: None
  - datalad_core (datalad 0.14.4):
    - distribution: datalad 0.14.4
    - load_error: None
    - module: datalad.metadata.extractors.datalad_core
    - version: None
  - datalad_rfc822 (datalad 0.14.4):
    - distribution: datalad 0.14.4
    - load_error: None
    - module: datalad.metadata.extractors.datalad_rfc822
    - version: None
  - exif (datalad 0.14.4):
    - distribution: datalad 0.14.4
    - load_error:
    - module: datalad.metadata.extractors.exif
  - frictionless_datapackage (datalad 0.14.4):
    - distribution: datalad 0.14.4
    - load_error: None
    - module: datalad.metadata.extractors.frictionless_datapackage
    - version: None
  - image (datalad 0.14.4):
    - distribution: datalad 0.14.4
    - load_error:
    - module: datalad.metadata.extractors.image
  - xmp (datalad 0.14.4):
    - distribution: datalad 0.14.4
    - load_error:
    - module: datalad.metadata.extractors.xmp
## metadata_indexers
## python
  - implementation: CPython
  - version: 3.9.2
## system
  - distribution:   10/10.0.19041/SP0/Multiprocessor Free
  - encoding:
    - default: utf-8
    - filesystem: utf-8
    - locale.prefered: cp1252
  - max_path_length: 286
  - name: Windows
  - release: 10
  - type: nt
  - version: 10.0.19041
```
</details>


Just to capture the cause of the test failure, they all seem to fail for the same reason: A non-fast-forward push 

```
======================================================================
ERROR: datalad.core.distributed.tests.test_push.test_push_git_annex_branch_many_paths_same_data
----------------------------------------------------------------------
Traceback (most recent call last):
  File "C:\Users\datalad\AppData\Local\Programs\Python\Python39\lib\site-packages\nose\case.py", line 198, in runTest
    self.test(*self.arg)
  File "C:\Users\datalad\repos\datalad\datalad\tests\utils.py", line 560, in _wrap_with_tree
    return t(*(arg + (d,)), **kw)
  File "C:\Users\datalad\repos\datalad\datalad\core\distributed\tests\test_push.py", line 840, in test_push_git_annex_branch_many_paths_same_data
    mk_push_target(ds, "target", str(path / "target"),
  File "C:\Users\datalad\repos\datalad\datalad\core\distributed\tests\test_push.py", line 124, in mk_push_target
    ds.repo.call_annex(['sync'])
  File "C:\Users\datalad\repos\datalad\datalad\support\annexrepo.py", line 1176, in call_annex
    return self._call_annex(
  File "C:\Users\datalad\repos\datalad\datalad\support\annexrepo.py", line 930, in _call_annex
    return runner.run(
  File "C:\Users\datalad\repos\datalad\datalad\cmd.py", line 168, in run
    raise CommandError(
datalad.support.exceptions.CommandError: CommandError: '"git" "-c" "diff.ignoreSubmodules=none" "annex" "sync" "-c" "annex.dotfiles=true" "-c" "annex.retry=3"' failed with exitcode 1 under C:\Users\datalad\AppData\Local\Temp\datalad_temp_tree_tvsqbbzd\ds [out: 'commit
On branch adjusted/dl-test-branch(unlocked)
nothing to commit, working tree clean
ok
pull target
(Merging into dl-test-branch...)
failed
(merging target/git-annex into git-annex...)
(recording state in git...)
push target
ok'] [err: 'From C:/Users/datalad/AppData/Local/Temp/datalad_temp_tree_tvsqbbzd/target
 * [new branch]      adjusted/dl-test-branch(unlocked) -> target/adjusted/dl-test-branch(unlocked)
 * [new branch]      dl-test-branch -> target/dl-test-branch
 * [new branch]      git-annex      -> target/git-annex
fatal: refusing to merge unrelated histories
To C:/Users/datalad/AppData/Local/Temp/datalad_temp_tree_tvsqbbzd/target
 * [new branch]      dl-test-branch -> synced/dl-test-branch
 * [new branch]      git-annex -> synced/git-annex
To C:/Users/datalad/AppData/Local/Temp/datalad_temp_tree_tvsqbbzd/target ! [rejected]        dl-test-branch -> dl-test-branch (non-fast-forward)error: failed to push some refs to 'C:/Users/datalad/AppData/Local/Temp/datalad_temp_tree_tvsqbbzd/target'hint: Updates were rejected because a pushed branch tip is behind its remotehint: counterpart. Check out this branch and integrate the remote changeshint: (e.g. 'git pull ...') before pushing again.hint: See the 'Note about fast-forwards' in 'git push --help' for details.sync: 1 failed']
```
```
======================================================================
ERROR: datalad.core.distributed.tests.test_push.test_push_matching
----------------------------------------------------------------------
Traceback (most recent call last):
  File "C:\Users\datalad\AppData\Local\Programs\Python\Python39\lib\site-packages\nose\case.py", line 198, in runTest
    self.test(*self.arg)
  File "C:\Users\datalad\repos\datalad\datalad\tests\utils.py", line 560, in _wrap_with_tree
    return t(*(arg + (d,)), **kw)
  File "C:\Users\datalad\repos\datalad\datalad\core\distributed\tests\test_push.py", line 861, in test_push_matching
    remote_ds = mk_push_target(ds, 'local', str(path / 'dssibling'),
  File "C:\Users\datalad\repos\datalad\datalad\core\distributed\tests\test_push.py", line 124, in mk_push_target
    ds.repo.call_annex(['sync'])
  File "C:\Users\datalad\repos\datalad\datalad\support\annexrepo.py", line 1176, in call_annex
    return self._call_annex(
  File "C:\Users\datalad\repos\datalad\datalad\support\annexrepo.py", line 930, in _call_annex
    return runner.run(
  File "C:\Users\datalad\repos\datalad\datalad\cmd.py", line 168, in run
    raise CommandError(
datalad.support.exceptions.CommandError: CommandError: '"git" "-c" "diff.ignoreSubmodules=none" "annex" "sync" "-c" "annex.dotfiles=true" "-c" "annex.retry=3"' failed with exitcode 1 under C:\Users\datalad\AppData\Local\Temp\datalad_temp_tree_x4_wiji8\ds [out: 'commit
On branch adjusted/dl-test-branch(unlocked)
nothing to commit, working tree clean
ok
pull local
(Merging into dl-test-branch...)
failed
(merging local/git-annex into git-annex...)
(recording state in git...)
push local
ok'] [err: 'From C:/Users/datalad/AppData/Local/Temp/datalad_temp_tree_x4_wiji8/dssibling
 * [new branch]      adjusted/dl-test-branch(unlocked) -> local/adjusted/dl-test-branch(unlocked)
 * [new branch]      dl-test-branch -> local/dl-test-branch
 * [new branch]      git-annex      -> local/git-annex
fatal: refusing to merge unrelated histories
To C:/Users/datalad/AppData/Local/Temp/datalad_temp_tree_x4_wiji8/dssibling
 * [new branch]      dl-test-branch -> synced/dl-test-branch
 * [new branch]      git-annex -> synced/git-annex
To C:/Users/datalad/AppData/Local/Temp/datalad_temp_tree_x4_wiji8/dssibling ! [rejected]        dl-test-branch -> dl-test-branch (non-fast-forward)error: failed to push some refs to 'C:/Users/datalad/AppData/Local/Temp/datalad_temp_tree_x4_wiji8/dssibling'hint: Updates were rejected because a pushed branch tip is behind its remotehint: counterpart. Check out this branch and integrate the remote changeshint: (e.g. 'git pull ...') before pushing again.hint: See the 'Note about fast-forwards' in 'git push --help' for details.sync: 1 failed']
```

```
======================================================================
ERROR: datalad.core.distributed.tests.test_push.test_gh1426
----------------------------------------------------------------------
Traceback (most recent call last):
  File "C:\Users\datalad\AppData\Local\Programs\Python\Python39\lib\site-packages\nose\case.py", line 198, in runTest
    self.test(*self.arg)
  File "C:\Users\datalad\repos\datalad\datalad\tests\utils.py", line 739, in _wrap_with_tempfile
    return t(*(arg + (filename,)), **kw)
  File "C:\Users\datalad\repos\datalad\datalad\tests\utils.py", line 739, in _wrap_with_tempfile
    return t(*(arg + (filename,)), **kw)
  File "C:\Users\datalad\repos\datalad\datalad\core\distributed\tests\test_push.py", line 543, in test_gh1426
    target = mk_push_target(
  File "C:\Users\datalad\repos\datalad\datalad\core\distributed\tests\test_push.py", line 124, in mk_push_target
    ds.repo.call_annex(['sync'])
  File "C:\Users\datalad\repos\datalad\datalad\support\annexrepo.py", line 1176, in call_annex
    return self._call_annex(
  File "C:\Users\datalad\repos\datalad\datalad\support\annexrepo.py", line 930, in _call_annex
    return runner.run(
  File "C:\Users\datalad\repos\datalad\datalad\cmd.py", line 168, in run
    raise CommandError(
datalad.support.exceptions.CommandError: CommandError: '"git" "-c" "diff.ignoreSubmodules=none" "annex" "sync" "-c" "annex.dotfiles=true" "-c" "annex.retry=3"' failed with exitcode 1 under C:\Users\datalad\AppData\Local\Temp\datalad_temp_k7ob17ro [out: 'commit
On branch adjusted/dl-test-branch(unlocked)
nothing to commit, working tree clean
ok
pull target
(Merging into dl-test-branch...)
failed
(merging target/git-annex into git-annex...)
(recording state in git...)
push target
ok'] [err: 'From C:/Users/datalad/AppData/Local/Temp/datalad_temp_r5qihdhm
 * [new branch]      adjusted/dl-test-branch(unlocked) -> target/adjusted/dl-test-branch(unlocked)
 * [new branch]      dl-test-branch -> target/dl-test-branch
 * [new branch]      git-annex      -> target/git-annex
fatal: refusing to merge unrelated histories
To C:/Users/datalad/AppData/Local/Temp/datalad_temp_r5qihdhm
 * [new branch]      dl-test-branch -> synced/dl-test-branch
 * [new branch]      git-annex -> synced/git-annex
To C:/Users/datalad/AppData/Local/Temp/datalad_temp_r5qihdhm ! [rejected]        dl-test-branch -> dl-test-branch (non-fast-forward)error: failed to push some refs to 'C:/Users/datalad/AppData/Local/Temp/datalad_temp_r5qihdhm'hint: Updates were rejected because a pushed branch tip is behind its remotehint: counterpart. Check out this branch and integrate the remote changeshint: (e.g. 'git pull ...') before pushing again.hint: See the 'Note about fast-forwards' in 'git push --help' for details.sync: 1 failed']
```

```
======================================================================
ERROR: datalad.core.distributed.tests.test_push.test_gh1763
----------------------------------------------------------------------
Traceback (most recent call last):
  File "C:\Users\datalad\AppData\Local\Programs\Python\Python39\lib\site-packages\nose\case.py", line 198, in runTest
    self.test(*self.arg)
  File "C:\Users\datalad\repos\datalad\datalad\tests\utils.py", line 739, in _wrap_with_tempfile
    return t(*(arg + (filename,)), **kw)
  File "C:\Users\datalad\repos\datalad\datalad\tests\utils.py", line 739, in _wrap_with_tempfile
    return t(*(arg + (filename,)), **kw)
  File "C:\Users\datalad\repos\datalad\datalad\tests\utils.py", line 739, in _wrap_with_tempfile
    return t(*(arg + (filename,)), **kw)
  File "C:\Users\datalad\repos\datalad\datalad\core\distributed\tests\test_push.py", line 591, in test_gh1763
    target1 = mk_push_target(src, 'target1', target1, bare=False)
  File "C:\Users\datalad\repos\datalad\datalad\core\distributed\tests\test_push.py", line 124, in mk_push_target
    ds.repo.call_annex(['sync'])
  File "C:\Users\datalad\repos\datalad\datalad\support\annexrepo.py", line 1176, in call_annex
    return self._call_annex(
  File "C:\Users\datalad\repos\datalad\datalad\support\annexrepo.py", line 930, in _call_annex
    return runner.run(
  File "C:\Users\datalad\repos\datalad\datalad\cmd.py", line 168, in run
    raise CommandError(
datalad.support.exceptions.CommandError: CommandError: '"git" "-c" "diff.ignoreSubmodules=none" "annex" "sync" "-c" "annex.dotfiles=true" "-c" "annex.retry=3"' failed with exitcode 1 under C:\Users\datalad\AppData\Local\Temp\datalad_temp_hhxwowzj [out: 'commit
On branch adjusted/dl-test-branch(unlocked)
nothing to commit, working tree clean
ok
pull target1
(Merging into dl-test-branch...)
failed
(merging target1/git-annex into git-annex...)
(recording state in git...)
push target1
ok'] [err: 'From C:/Users/datalad/AppData/Local/Temp/datalad_temp_n4oob8np
 * [new branch]      adjusted/dl-test-branch(unlocked) -> target1/adjusted/dl-test-branch(unlocked)
 * [new branch]      dl-test-branch -> target1/dl-test-branch
 * [new branch]      git-annex      -> target1/git-annex
fatal: refusing to merge unrelated histories
To C:/Users/datalad/AppData/Local/Temp/datalad_temp_n4oob8np
 * [new branch]      dl-test-branch -> synced/dl-test-branch
 * [new branch]      git-annex -> synced/git-annex
To C:/Users/datalad/AppData/Local/Temp/datalad_temp_n4oob8np ! [rejected]        dl-test-branch -> dl-test-branch (non-fast-forward)error: failed to push some refs to 'C:/Users/datalad/AppData/Local/Temp/datalad_temp_n4oob8np'hint: Updates were rejected because a pushed branch tip is behind its remotehint: counterpart. Check out this branch and integrate the remote changeshint: (e.g. 'git pull ...') before pushing again.hint: See the 'Note about fast-forwards' in 'git push --help' for details.sync: 1 failed']

----------------------------------------------------------------------
```

Looking into the repositories, the commits in ``dl-test-branch`` of the source and the target dataset are indeed different:

```

datalad@latitude-e7440 MINGW64 /tmp/datalad_temp_tree_5lvpt4_p/ds (adjusted/dl-test-branch(unlocked))
$ datalad siblings
.: here(+) [git]
.: target(?) [C:/Users/datalad/AppData/Local/Temp/datalad_temp_tree_5lvpt4_p/target (git)]

datalad@latitude-e7440 MINGW64 /tmp/datalad_temp_tree_5lvpt4_p/ds (adjusted/dl-test-branch(unlocked))
$ git -C C:/Users/datalad/AppData/Local/Temp/datalad_temp_tree_5lvpt4_p/target log --oneline
502d249 (HEAD -> adjusted/dl-test-branch(unlocked)) git-annex adjusted branch
561afe5 (dl-test-branch, refs/basis/adjusted/dl-test-branch(unlocked)) commit before entering adjusted branch

datalad@latitude-e7440 MINGW64 /tmp/datalad_temp_tree_5lvpt4_p/ds (adjusted/dl-test-branch(unlocked))
$ git log --oneline
06bb679 (HEAD -> adjusted/dl-test-branch(unlocked)) git-annex adjusted branch
7074b93 (target/synced/dl-test-branch, synced/dl-test-branch, dl-test-branch, refs/basis/adjusted/dl-test-branch(unlocked)) [DATALAD] Recorded changes
3166043 [DATALAD] new dataset
5bdd3ce commit before entering adjusted branch
```

**Comment by mih** [comment]
So from the logs it seems that it isn't `push()` that is failing, but `git annex sync`. Here is a relevant comment near that call

```
        # maximum complication
        # the target repo already has a commit that is unrelated
        # to the source repo, because it has built a reference
        # commit for the managed branch.
        # the only sane approach is to let git-annex establish a shared
        # history
 ```

So the situation is:

- there is a single repository with some files annexed (non-bare, in adjusted(unlock) mode)
- a new repository is created (`git init`, `git annex init`) that ends up in adjusted(unlock) mode, due to windows)
- the new repo get's a `receive.denyCurrentBranch=updateInstead` config in its local config
- the previously existing repo gets the new one added as a remote

The subsequent call to `git annex sync` fails. The error output shown above is caused by a `git push` issued by `git annex sync`, not by datalad. IOW the failure is in the test setup already.

It seems `git annex sync` is no longer able to reconciliate the history of the original branch with the non-history of the new repository.

**Comment by adswa** [comment]
Ahh thanks, I see! Has anyone filed an issue/question at git-annex?

**Comment by yarikoptic** [comment]
AFAIK not yet -- my windows foo of `git annex get` via ssh failed me so I didn't get to reproduce it etc.  If you could @adswa to file it -- would be great, although may be @joeyh can already see what to fix from the @mih description [above](https://github.com/datalad/datalad/issues/5811#issuecomment-887424283).

**Comment by adswa** [comment]
I haven't filed anything against git-annex (sorry - I always feel like I am too clueless for any issue at hand to file a meaningful bug report), but I was thinking whether it may actually be our test setup that should get a fix. I think a patch like this could work:

```diff
diff --git a/datalad/core/distributed/tests/test_push.py b/datalad/core/distributed/tests/test_push.py
index 1a40371c6..d35248923 100644
--- a/datalad/core/distributed/tests/test_push.py
+++ b/datalad/core/distributed/tests/test_push.py
@@ -121,7 +121,10 @@ def mk_push_target(ds, name, path, annex=True, bare=True):
         # commit for the managed branch.
         # the only sane approach is to let git-annex establish a shared
         # history
-        ds.repo.call_annex(['sync'])
+        if AnnexRepo.git_annex_version > "8.20210631":
+            ds.repo.call_annex(['sync', '--allow-unrelated-histories'])
+        else:
+            ds.repo.call_annex(['sync'])
         ds.repo.call_annex(['sync', '--cleanup'])
     return target
```

It uses the ``allow-unrelated-histories`` flag of ``sync`` that was added two weeks ago in http://source.git-annex.branchable.com/?p=source.git;a=commit;h=3d50b47dede6e4e94232718f4c460cd19147dcfb

**Comment by mih** [comment]
Fixed via #5855
